community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/files/LogFilesBuilder.java

```diff
@@ -309,6 +310,37 @@ private LongSupplier lastCommittedIdSupplier() throws IOException
         }
     }
 
+    private LongSupplier committingIdSupplier() throws IOException
+    {
+        if ( transactionIdStore != null )
+        {
+            return () -> transactionIdStore.committingTransactionId();
```

Comment:
I think we should use method references instead, because that captures the target object instance, instead of capturing this builder object and referencing through the `transactionIdStore` field on every call. This also applies to the dependency resolver based lambda bellow, and in the `lastCommittedIdSupplier` method above.

Reply:
updated to method references in mentioned cases, except where its based on dependency resolver since that is lazy case and we do not have TransactionIdStore during construction time yet